src/Features/markdown/Readme.tsx

```diff
@@ -0,0 +1,228 @@
+import { useEffect, useState } from "react";
+import { useFile } from "@/hooks";
+import MarkdownEditor from "@/Features/markdown/Editor";
+import { templates } from "@/Features/markdown/templates";
+import {
+  DragDropContext,
+  Draggable,
+  Droppable,
+  OnDragEndResponder,
+} from "@hello-pangea/dnd";
+import cx from "clsx";
+import classes from "@/Layout/Navbar/components/ungrouped.module.css";
+import MarkdownPreview from "@uiw/react-markdown-preview";
+import remarkMath from "remark-math";
+import rehypeKatex from "rehype-katex";
+import rehypeRaw from "rehype-raw";
+
+type Part = {
+  title: string;
+  template: string;
+};
+
+// TODO: Build more advanced templates like Awsome Readme & some cleanup / reusability
+
+const Readme = () => {
+  const [showPreview, setShowPreview] = useState(true);
+  const [showEditor, setShowEditor] = useState(true);
+  const [opened, setOpened] = useState(false);
+
+  const [parts, setParts] = useState<Part[]>([
+    {
+      title: "Header",
+      template: templates.header as string,
+    },
+    {
+      title: "Acknowledgements",
+      template: templates.acknowledgements as string,
+    },
+  ]);
+
+  const [activeIndex, setActiveIndex] = useState(0);
+  const updatePart = (template: string) => {
+    setParts((p) =>
+      p.map((part, index) =>
+        index === activeIndex ? { ...part, template } : part
+      )
+    );
+  };
+  const activePart = parts[activeIndex];
+
+  const combineParts = () => parts.map((part) => part.template).join("\n");
+
+  const { file, setFile, openFile, saveFile } = useFile({
+    initialFile: combineParts(),
+  });
+
+  useEffect(() => {
+    setFile(combineParts());
+  }, [parts]);
+
+  const onDragEnd: OnDragEndResponder = (res) => {
+    if (res.destination?.index === res.source.index) return;
+    const items = [...parts];
+    const [reorderedItem] = items.splice(res.source.index, 1);
+    items.splice(res.destination!.index, 0, reorderedItem);
+    setParts(items);
+  };
+
+  return (
+    <Stack
+      style={{
+        height: "100%",
+      }}
+    >
+      <Group>
+        <Button onClick={openFile}>Open md file</Button>
+        <Button onClick={saveFile}>Save md file</Button>
+        <Button
+          onClick={() => setShowPreview((p) => !p)}
+          disabled={!showEditor}
+        >
+          {showPreview ? "Hide" : "Show"} preview
+        </Button>
+        <Button
+          onClick={() => setShowEditor((p) => !p)}
+          disabled={!showPreview}
+        >
+          {showEditor ? "Hide" : "Show"} editor
+        </Button>
+        {activeIndex ? (
+          <Button
+            onClick={() => {
+              setParts((prev) =>
+                prev.filter((_, index) => index !== activeIndex)
+              );
+              setActiveIndex(0);
+            }}
+          >
+            Remove
+          </Button>
+        ) : null}
+      </Group>
+      <div style={{ width: "100%", height: "95%", display: "flex" }}>
```

Comment:
PS: Mantine has `Flex` or `Group`, if you just want a div with display:flex. 
you can also do `<Box display="flex">` for simplicity if you want.